framework/Models/HybridModel.py

```diff
@@ -0,0 +1,621 @@
+        selectedKwargs['SampledVars'][key] = kwargs["SampledVars"][key]
+        selectedKwargs['SampledVarsPb'][key] = kwargs["SampledVarsPb"][key] if 'SampledVarsPb' in kwargs.keys() else 1.0
+    return selectedKwargs
+
+  def createNewInput(self,myInput,samplerType,**kwargs):
+    """
+      This function will return a new input to be submitted to the model, it is called by the sampler.
+      @ In, myInput, list, the inputs (list) to start from to generate the new one
+      @ In, samplerType, string, is the type of sampler that is calling to generate a new input
+      @ In, **kwargs, dict,  is a dictionary that contains the information coming from the sampler,
+           a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
+      @ Out, newInputs, dict, dict that returns the new inputs for each sub-model
+    """
+    self.raiseADebug("Create New Input")
+    if self.romValid:
+      identifier = kwargs['prefix']
+      newKwargs = {'prefix':identifier}
+      for romName in self.romsDictionary.keys():
+        newKwargs[romName] = self.__selectInputSubset(romName, kwargs)
+        newKwargs[romName]['prefix'] = romName+utils.returnIdSeparator()+identifier
+        newKwargs[romName]['uniqueHandler'] = self.name+identifier
+    else:
+      newKwargs = copy.deepcopy(kwargs)
+
+    if self.modelInstance.type == 'Code':
+      codeInput = []
+      romInput = []
+      for elem in myInput:
+        if isinstance(elem, Files.File):
+          codeInput.append(elem)
+        elif elem.type in ['PointSet', 'HistorySet']:
+          romInput.append(elem)
+        else:
+          self.raiseAnError(IOError, "The type of input ", elem.name, " can not be accepted!")
+      if self.romValid:
+        return (romInput, samplerType, newKwargs)
+      else:
+        return (codeInput, samplerType, newKwargs)
+
+    return (myInput, samplerType, newKwargs)
+
+  def trainRom(self, samplerType, kwargs):
+    """
+      This function will train all ROMs if they are not converged
+      @ In, None
+      @ Out, None
+    """
+    self.raiseADebug("Start to train roms")
+    self.raiseADebug("Current sample: ", self.counter)
+    for romInfo in self.romsDictionary.values():
+      # reset the rom
+      romInfo['Instance'].amITrained = False
+      # always train the rom even if the rom is converged, we assume the cross validation and rom train are relative cheap
+      outputMetrics = self.cvInstance.evaluateSample([romInfo['Instance'], self.tempTargetEvaluation], samplerType, kwargs)[1]
+      converged = self.isRomConverged(outputMetrics)
+      romInfo['Converged'] = converged
+      if converged:
+        romInfo['Instance'].train(self.tempTargetEvaluation)
+        self.raiseADebug("ROM ", romInfo['Instance'].name, " is converged!")
+
+  def isRomConverged(self, outputDict):
+    """
+      This function will check the convergence of rom
+      @ In, outputDict, dict, dictionary contains the metric information
```

Comment:
the topology of the dict should be reported (what does it contain and how?)
foR example,
{'targetName':metricValue}

Reply:
added